python/rola/utils/gateway.py

```diff
@@ -0,0 +1,17 @@
+import requests
+from rola.config.network import NETWORKS
+
+
+def get_entity_owner(
```

Comment:
How about we give users of this library more flexibility in here. More specifically, how about:

* We allow them to use their own custom gateways if they want to.
* We allow them to use their own services if they want to.

I would recommend something like the following:

```py
from abc import ABC

class MetadataProvider(ABC):
    @abstractmethod
    def entity_owner(self, address: str) -> list[PublicKeyHash]:
        pass

class GatewayMetadataProvider(MetadataProvider):
    base_url: str

    def __init__(self, base_url: str = "https://mainnet.radixdlt.com") -> None:
        self.base_url = base_url

    @classmethod
    def for_mainnet(cls) -> GatewayMetadataProvider:
        return cls("https://mainnet.radixdlt.com")
    
    @classmethod
    def for_stokenet(cls) -> GatewayMetadataProvider:
        return cls("https://stokenet.radixdlt.com")

    def entity_owner(self, address: str) -> list[PublicKeyHash]:
        # The implementation here
        pass
```

This means that users can specify if they want to use their own gateway or different network and also have the option to go for the simple options given in the various class methods. Additionally, it allows users to not use the gateway API if they do not want to (e.g., use the Core API) by implementing a class that implements the `MetadataProvider` interface.

Reply:
I do like it. applying this change

Reply:
done